### ISSUE TITLE
Support extensions with bundled dependent libraries

### ISSUE THREAD

**Issue by daa** [mbachry/exxo#9]
As I see in patch in function `import_solib` you extract extension shared object file and load it, however it's insufficient if module is linked with bundled libraries. This means that multiple `manylinux1` wheels are not supported, for example you may look at `numpy` - there extension modules have `RPATH` pointing to package root `.libs` directory:

```
 $ readelf -a numpy/core/multiarray.so | grep RPATH
 0x000000000000000f (RPATH)              Library rpath: [$ORIGIN/../.libs]
```

And after extraction that module will not find required library at `RPATH` and will look at system-wide libraries, where it may not find what it wants.

If I'm wrong and `exxo` handles this issue, please correct me.

One solution may be extracting all `.so` files keeping their relative structure, another - recursively determine all dependencies and extract only them. Or you may extract them to `RUNPATH` directory of `pyrun`.

Also I'd like to suggest creating import hook in Python and adding it to `sys.meta_path` instead of patching `zipimport`.

**Comment by mbachry** [comment]
Thanks for pointing it out. Numpy worked for me before, but it looks like they started to deliver manylinux wheels recently. I'll try to tackle it and I think I'll follow your idea of unpacking bundled libraries using `RPATH`  section.

The suggestion of moving code to an import hook is spot on! This may be a good opportunity, as handling RPATH stuff in C code can get hairy pretty quick. Thanks a lot!

**Comment by mbachry** [comment]
I added initial support for RPATH (test suite is a big ugly at the moment though). It's enough to make numpy work. I plan to revisit and clean up RPATH when I'll implement bundling dependant libraries when building a binary (eg. libxml2 for lxml). Feel free to open a new issue if you see something else broken.

I also moved lot of C code to pure Python meta_path importer. Thanks again for inspiration here.
